### PR TITLE
Allow the Debug system plugin to write executed query to a file

### PR DESCRIPTION
**Pull request by alikon**
### Purpose

Speed up the process of benchmarking  SQL queries executed by a Joomla! page with tools like mysqlslap, pgbench, etc..., writing a file with the extecuted query for the page visited
### How it works

Add a parameter (disabled for default) to the logging tab: Log extecuted SQL
pluginsystemdebug

When enabled it write a file to the logs folder named: 
sitename_alias+id for frontend 
sitename_option_view_layout for administration side

### CODE REVIEW

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -634,6 +634,11 @@ protected function displayProfileInformation()
 						'PLG_DEBUG_QUERIES_TIME',
 						sprintf('<span class="label ' . $labelClass . '">%.1f&nbsp;ms</span>', $totalQueryTime)
 					) . '</div>';
+
+				if ($this->params->get('log-executed-sql', '0'))
+				{
+					$this->write2file();
```
Comment (alikon) [inline plugins/system/debug/debug.php]: done

File: plugins/system/debug/debug.php [hunk 2]
```diff
@@ -1743,4 +1748,86 @@ protected function displayLogs()
...
+		if (JFile::exists($file))
+		{
+			JFile::delete($file);
+		}
+
+		// Write new file. First try with JFile.
+		$result = JFile::write($file, $current);
+
+		// In case JFile used FTP but direct access could help.
+		if (!$result)
```
Comment (alikon) [inline plugins/system/debug/debug.php]: block removed

**Comment by roland-d** [comment]
@test: functionality works as expected, however the option Log almost everything is gone. Please update your PR against latest staging.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5445">issues.joomla.org/joomla-cms/5445</a>.</sub>

**Comment by alikon** [comment]
@roland-d  updated

**Comment by roland-d** [comment]
@alikon Sorry to say but the PR can't be applied, I get the following errors:

rror: patch failed: administrator/language/en-GB/en-GB.plg_system_debug.ini:1
error: administrator/language/en-GB/en-GB.plg_system_debug.ini: patch does not apply
error: patch failed: plugins/system/debug/debug.php:3
error: plugins/system/debug/debug.php: patch does not apply
error: patch failed: plugins/system/debug/debug.xml:3
error: plugins/system/debug/debug.xml: patch does not apply

This means your files our out-of-sync with the current staging.

File: .gitignore [hunk 3]
```diff
@@ -95,3 +95,4 @@ Desktop.ini
 /libraries/vendor/symfony/yaml/Symfony/Component/Yaml/composer.json
 /libraries/vendor/symfony/yaml/Symfony/Component/Yaml/phpunit.xml.dist
 /libraries/vendor/symfony/yaml/Symfony/Component/Yaml/README.md
+/nbproject/private/
```
Comment (alikon) [inline .gitignore]: yes

**Comment by alikon** [comment]
close as per #6202

**Pull request closed without merging**